Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc

```diff
@@ -0,0 +1,308 @@
+      t_TracksProp = itr->tracksProp_;
+      t_TracksSaved = itr->tracksSaved_;
+      t_TracksLoose = itr->tracksLoose_;
+      t_TracksTight = itr->tracksTight_;
+      t_allvertex = itr->allvertex_;
+      t_ietaAll = &itr->ietaAll_;
+      t_ietaGood = &itr->ietaGood_;
+      t_trackType = &itr->trackType_;
+      t_hltbits = &itr->hltbits_;
+      tree2->Fill();
+    }
+  } else {
+    edm::LogVerbatim("HcalIsoTrack") << "Cannot find HcalIsoTrkEventVariablesCollections";
+  }
+}
+
+void HcalIsoTrackAnalyzer::beginJob() {
+  tree = fs->make<TTree>("CalibTree", "CalibTree");
+
+  tree->Branch("t_Run", &t_Run, "t_Run/I");
+  tree->Branch("t_Event", &t_Event, "t_Event/I");
+  tree->Branch("t_DataType", &t_DataType, "t_DataType/I");
+  tree->Branch("t_ieta", &t_ieta, "t_ieta/I");
+  tree->Branch("t_iphi", &t_iphi, "t_iphi/I");
+  tree->Branch("t_EventWeight", &t_EventWeight, "t_EventWeight/D");
+  tree->Branch("t_nVtx", &t_nVtx, "t_nVtx/I");
+  tree->Branch("t_nTrk", &t_nTrk, "t_nTrk/I");
+  tree->Branch("t_goodPV", &t_goodPV, "t_goodPV/I");
+  tree->Branch("t_l1pt", &t_l1pt, "t_l1pt/D");
+  tree->Branch("t_l1eta", &t_l1eta, "t_l1eta/D");
+  tree->Branch("t_l1phi", &t_l1phi, "t_l1phi/D");
+  tree->Branch("t_l3pt", &t_l3pt, "t_l3pt/D");
+  tree->Branch("t_l3eta", &t_l3eta, "t_l3eta/D");
+  tree->Branch("t_l3phi", &t_l3phi, "t_l3phi/D");
+  tree->Branch("t_p", &t_p, "t_p/D");
+  tree->Branch("t_pt", &t_pt, "t_pt/D");
+  tree->Branch("t_phi", &t_phi, "t_phi/D");
+  tree->Branch("t_mindR1", &t_mindR1, "t_mindR1/D");
+  tree->Branch("t_mindR2", &t_mindR2, "t_mindR2/D");
+  tree->Branch("t_eMipDR", &t_eMipDR, "t_eMipDR/D");
+  tree->Branch("t_eMipDR2", &t_eMipDR2, "t_eMipDR2/D");
+  tree->Branch("t_eMipDR3", &t_eMipDR3, "t_eMipDR3/D");
+  tree->Branch("t_eMipDR4", &t_eMipDR4, "t_eMipDR4/D");
+  tree->Branch("t_eMipDR5", &t_eMipDR5, "t_eMipDR5/D");
+  tree->Branch("t_eHcal", &t_eHcal, "t_eHcal/D");
+  tree->Branch("t_eHcal10", &t_eHcal10, "t_eHcal10/D");
+  tree->Branch("t_eHcal30", &t_eHcal30, "t_eHcal30/D");
+  tree->Branch("t_hmaxNearP", &t_hmaxNearP, "t_hmaxNearP/D");
+  tree->Branch("t_emaxNearP", &t_emaxNearP, "t_emaxNearP/D");
+  tree->Branch("t_eAnnular", &t_eAnnular, "t_eAnnular/D");
+  tree->Branch("t_hAnnular", &t_hAnnular, "t_hAnnular/D");
+  tree->Branch("t_rhoh", &t_rhoh, "t_rhoh/D");
+  tree->Branch("t_selectTk", &t_selectTk, "t_selectTk/O");
+  tree->Branch("t_qltyFlag", &t_qltyFlag, "t_qltyFlag/O");
+  tree->Branch("t_qltyMissFlag", &t_qltyMissFlag, "t_qltyMissFlag/O");
+  tree->Branch("t_qltyPVFlag", &t_qltyPVFlag, "t_qltyPVFlag/O");
+  tree->Branch("t_gentrackP", &t_gentrackP, "t_gentrackP/D");
+
+  t_DetIds = new std::vector<unsigned int>();
+  t_DetIds1 = new std::vector<unsigned int>();
+  t_DetIds3 = new std::vector<unsigned int>();
+  t_HitEnergies = new std::vector<double>();
+  t_HitEnergies1 = new std::vector<double>();
+  t_HitEnergies3 = new std::vector<double>();
+  t_trgbits = new std::vector<bool>();
```

Comment:
I think `new` is also discouraged, use `std::make_unique` instead or at least make sure that if you stay with the new there is a `delete` in the end to free the memory and avoid memory leaks. But my understanding is that CMS prefers to use the unique pointers instead

Reply:
ok, it that case, can you just make sure that everything is `delete`-ed in the end?

Reply:
I am removing the use of "new"